dwave/cloud/config.py

```diff
@@ -255,94 +408,92 @@ def get_default_config():
 
 def load_config(config_file=None, profile=None, client=None,
                 endpoint=None, token=None, solver=None, proxy=None):
-    """Load D-Wave cloud client configuration from ``config_file`` (either
-    explicitly specified, or auto-detected) for ``profile``, but override the
-    values from config file with values defined in process environment, and
-    override those with values specified with keyword arguments.
-
-    Config file uses the standard Windows INI format, with one profile per
-    section. A section providing default values for all other sections is called
-    ``defaults``.
-
-    If the location of ``config_file`` is not specified, an auto-detection is
-    performed (looking first for ``dwave.conf`` in process' current working
-    directory, then in user-local config directories, and finally in system-wide
-    config dirs). For details on format and location detection, see
-    :func:`load_config_from_files`.
-
-    If location of ``config_file`` is explicitly specified (via arguments or
-    environment variable), but the file does not exits, or is not readable,
-    config loading will fail with
-    :exc:`~dwave.cloud.exceptions.ConfigFileReadError`. Config loading will fail
-    with :exc:`~dwave.cloud.exceptions.ConfigFileParseError` if file is
-    readable, but it's not a valid config file.
-
-    Similarly, if ``profile`` is explicitly specified (via arguments or
-    environment variable), config loading will fail with :exc:`ValueError` if
-    that profile is not present in the config loaded. If config file is not
-    specified explicitly, nor detected on file system, not defined via
-    environment, resulting in an empty config, explicit profile selection will
-    also fail.
-
-    If profile is not explicitly specified, selection of profile is described
-    under ``profile`` argument below.
+    """Load D-Wave Cloud Client configuration based on a configuration file.
+
+    Configuration files comply with standard Windows INI-like format,
+    parsable with Python's :mod:`configparser`. An optional ``defaults`` section
+    provides default key/values for all other sections.
+
+    Configuration values can be specified in multiple ways, ranked in the following
+    order (with 1 the highest ranked):
```

Comment:
The correct order is:
- kwargs
- env
- config profile
- config defaults

But I'm not sure we need to separate config defaults from profile - it's just a profile loaded from a (set of) config file(s). And config loading mechanism uses [defaults]-values for missing profile values.